### PR TITLE
HDDS-15985. Add new Overview page for Ozone Manager

### PR DESCRIPTION
**Pull request by spacemonkd**
## What changes were proposed in this pull request?
HDDS-15985. Add new Overview page for Ozone Manager

Please describe your PR in detail:
This PR adds the new React based Overview page for Ozone Manager

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-15985

## How was this patch tested?
Patch was tested manually via mocked DB.

https://github.com/user-attachments/assets/2e289b56-9bed-405e-ade8-ff46d7c2aee8

Output of "Copy Arguments" button at the end:
```
<configuration>
  <property>
    <name>-Dorg.apache.ratis.thirdparty.io.netty.allocator.useCacheForAllThreads</name>
    <value>false</value>
  </property>
  <property>
    <name>-XX:+PrintGCDetails</name>
    <value>Enabled</value>
  </property>
</configuration>
```

### CODE REVIEW

**Comment by spacemonkd** [comment]
@chihsuan would you like to take a look as well?

**Comment by chihsuan** [comment]
> @chihsuan would you like to take a look as well?

Thanks for the ping! Sure, I'll take a look. 🙂

File: ozone-ui/packages/om/src/api/overview.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+   */
+  leaderElectionCount: 'ratis:name=ratis.leader_election.*electionCount',
+  leaderElectionElapsed: 'ratis:name=ratis.leader_election.*lastLeaderElectionElapsedTime',
+} as const;
+
+/* --------------------------------- Beans ---------------------------------- */
+
+export interface OzoneManagerInfoBean {
+  RpcPort: string;
+  RatisRoles: string;
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/overview.ts]: I tried to run this against a real OM, and found the Overview page renders blank. It seems that `RatisRoles` in the JMX bean is an array of tuples, not a string:

https://github.com/apache/ozone/blob/ebe71f64b754ffe6b6a2965282833d4bd292879d/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMMXBean.java#L32

 I would change `RatisRoles: string` to `string[][]` and update the mock.

File: ozone-ui/packages/om/src/api/overview.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+}
+
+/* -------------------------------- Parsers --------------------------------- */
+
+/**
+ * Parse the OM `RatisRoles` string, e.g.
+ * `{ HostName: h1 | Node-Id: om1 | Ratis-Port : 9872 | Role: FOLLOWER } {...}`.
+ */
+export function parseRatisRoles(raw: string, currentNodeId?: string): RatisRole[] {
+  const groups = raw?.match(/\{[^}]*\}/g) ?? [];
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/overview.ts]: Same issue as above: with a real OM, `raw.match(...)` throws `TypeError: raw.match` is not a function. Could we parse the tuple array here?

File: ozone-ui/packages/om/vite.config.ts [hunk 3]
```diff
@@ -74,6 +74,11 @@ export default defineConfig({
       '/api': {
         target: 'http://localhost:9862',
       },
+      // JMX endpoint — proxied to the json-server mock in dev (see mock/server.cjs).
+      '/jmx': {
+        target: 'http://localhost:9878',
```
Comment (chihsuan) [inline ozone-ui/packages/om/vite.config.ts]: I hit a port conflict here when trying to run this against a real OM. `9878` is also the S3 Gateway port in the Compose cluster, and pointing at a real OM requires editing this file.

Would it make sense to support an env override? e.g. `target: process.env.OM_JMX_TARGET ?? 'http://localhost:9878'`?

File: ozone-ui/packages/om/src/api/overview.ts [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+  if (!xmx) {
+    return 'Not set';
+  }
+  const match = xmx.slice(4).match(/^(\d+)\s*([kKmMgG])?/);
+  if (!match) {
+    return xmx.slice(4);
+  }
+  const size = Number(match[1]);
+  const unit = (match[2] ?? 'B').toUpperCase();
+  const megabytes = unit === 'G' ? size * 1024 : unit === 'K' ? Math.round(size / 1024) : size;
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/overview.ts]: I noticed that the JVM interprets an `-Xmx` value without a suffix as bytes, but the `B` case currently labels the raw value as MB. For example, `-Xmx2511000000` would be displayed as `2,511,000,000 MB`.

I think we should divide byte values by 1024x1024 before displaying them as MB and could probably consider using a switch/case for better readability

```js
  function formatHeap(xmx: string | undefined): string {
    if (!xmx) {
      return 'Not set';
    }

    const match = xmx.slice(4).match(/^(\d+)\s*([kKmMgG])?/);
    if (!match) {
      return xmx.slice(4);
    }

    const size = Number(match[1]);
    const unit = (match[2] ?? 'B').toUpperCase();

    let megabytes: number;
    switch (unit) {
      case 'G':
        megabytes = size * 1024;
        break;
      case 'M':
        megabytes = size;
        break;
      case 'K':
        megabytes = size / 1024;
        break;
      default:
        megabytes = size / (1024 * 1024);
        break;
    }

    return `${Math.round(megabytes).toLocaleString('en-US')} MB`;
  }
```

File: ozone-ui/packages/om/src/pages/Overview/sections/RolesSection.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+];
+
+/** "Ozone Manager Roles" HA table. Sourced from the OM ServerRuntime bean. */
+export const RolesSection: React.FC<SectionProps> = ({ refreshToken }) => {
+  const {
+    data: omInfo,
+    loading,
+    error,
+  } = useJmxBean<OzoneManagerInfoBean>(JMX_QUERY.omInfo, refreshToken);
+  const { data: ratis } = useJmxBean<RatisServerBean>(JMX_QUERY.ratisServer, refreshToken);
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/pages/Overview/sections/RolesSection.tsx]: These hooks only consume data, so failures in the Ratis or leader-election queries appear as `—` or missing fields. Could we handle these query errors, at least by showing a partial-data warning? Or do we plan to do this in a follow-up?

File: ozone-ui/packages/om/src/api/jmx.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    });
+    cache.set(qry, pending as Promise<unknown[]>);
+  }
+  return pending;
+}
+
+/** Fetch a single MBean for a query (the first bean), or `undefined`. */
+export async function fetchJmxBean<T>(qry: string): Promise<T | undefined> {
+  const beans = await fetchJmxBeans<T>(qry);
+  return beans[0];
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/jmx.ts]: When no MBean matches a query, the JMX servlet can return 200 OK with `{"beans":[]}`.

I think this might happen when an MBean isn't yet registered, during service reinitialization, or when the UI and OM expose different MBean versions. Would it be worth representing “no matching bean” as an explicit empty state and showing a message such as No JMX data available?

File: ozone-ui/packages/om/src/pages/Overview/sections/RolesSection.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  padding: '4px 8px 8px',
+};
+
+const columns: TableColumnsType<RatisRole> = [
+  {
+    title: 'Host Name',
+    dataIndex: 'hostName',
+    key: 'hostName',
+    render: (hostName: string, row) => (
+      <TextLink href="#" style={{ fontWeight: row.isCurrent ? 600 : undefined }}>
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/pages/Overview/sections/RolesSection.tsx]: Is `href="#"` intentional here, or a placeholder for now?

If the real link is planned for a follow-up, would plain text work better for now so it doesn't read as clickable?

Image

File: ozone-ui/packages/om/src/api/jmx.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+/** In-flight / resolved query cache, keyed by the JMX query string. */
+const cache = new Map<string, Promise<unknown[]>>();
+
+/**
+ * Fetch MBeans for a query, sharing a single request across all callers that ask
+ * for the same query. Failed requests are evicted so they can be retried.
+ */
+export function fetchJmxBeans<T>(qry: string): Promise<T[]> {
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/jmx.ts]: Since this is the first data-fetching page in the new UI, though, do we want to consider a library like `TanStack Query` for this layer?

As more OM and SCM pages add API queries, this may become difficult to maintain and introduce edge cases such as the refresh/cache race.

Other ASF React UIs already use this approach, including the [Apache Airflow UI](https://github.com/apache/airflow/blob/main/airflow-core/src/airflow/ui/package.json) and [Apache APISIX Dashboard](https://github.com/apache/apisix-dashboard/blob/master/package.json). Since this introduces a dependency, it could also be tracked as a follow-up if it is outside this PR’s scope.

File: ozone-ui/packages/om/src/pages/Overview/SectionBody.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  /** Number of skeleton rows to show while loading. Defaults to 2. */
+  skeletonRows?: number;
+  children: React.ReactNode;
+}
+
+/**
+ * Renders a section's async state: a skeleton while loading, an error alert on
+ * failure, or the resolved content.
+ */
+export const SectionBody: React.FC<SectionBodyProps> = ({
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/pages/Overview/SectionBody.tsx]: `SectionBody` handles loading and errors, but not the empty state here. I think we can consider centralizing it so that state management is all in this component.

For example, adding an `isEmpty` prop rendering a small "No data" placeholder would make
this visible with a minimal diff.

Longer term, if we adopt TanStack Query (see the other comment), its Suspense mode
would restructure this nicely, like below:

```tsx
<ErrorBoundary fallback={<Alert ... />}>
  <Suspense fallback={<Skeleton rows={2} />}>
    <InstanceDetailsContent />  {/* useSuspenseQuery inside, data typed non-null */}
  </Suspense>
</ErrorBoundary>
```

Not asking for that in this PR, just sharing some thoughts. 🙂

**Review by chihsuan — COMMENTED** [review]
Nice work overall, @spacemonkd! The page structure is clean, and the mock setup makes local development easy.

I tested it with both the mock and a real OM and found a few issues. Please see the inline comments.

I also wanted to share a few broader suggestions. None of these need to be addressed in this PR: 🙂 

1. **Unit tests**: I understand that the current diff is already quite large 😅. the pure parsers in `overview.ts` should be easy to cover.
2. **JMX as a UI contract**: The UI currently depends directly on JMX bean shapes, which can change silently when the Java implementation evolves. It might be worth considering documenting or validating the bean shapes consumed by the UI. Longer term, a dedicated REST endpoint could provide a more stable contract.
3. **Data fetching**: It would be useful to decide early whether the new UIs should adopt a server-state library such as TanStack Query. I left an inline comment with more context.
4. **Storybook**: As the shared component library grows, what do you think about setting up Storybook? I think this could help develop, review, and document shared components in isolation.

Happy to help with any of these follow-ups.

**Comment by spacemonkd** [comment]
Thanks @chihsuan for the extensive review. Sorry I got late in updating, I was thinking of adding TanStack Query. The latest changes add it as well and fixes the comments you gave.
For the storybook - you gave a good point, I am planning to add a new task at the end once we have all the components finalized.
I think that will be a good documentation! Thanks for this suggestion.

Could you please take a look again?

File: ozone-ui/packages/om/src/api/overview.ts [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+
+/**
+ * Parse the OM `RatisRoles` bean — an array of
+ * `[hostName, nodeId, ratisPort, role, leaderReadiness]` tuples. Rows that don't
+ * carry at least the first four fields (e.g. the single-element error row the
+ * bean returns when there is no leader) are skipped.
+ */
+export function parseRatisRoles(rows: string[][] | undefined, currentNodeId?: string): RatisRole[] {
+  return (rows ?? [])
+    .filter((row) => Array.isArray(row) && row.length >= 4)
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/overview.ts:136]: When there is no leader, the bean returns a one-element message row, and this filter drops it, so the table is empty. Should we show the message?  I noticed the old om-overview.html rendered that message.

File: ozone-ui/packages/om/src/pages/Overview/sections/JvmSection.tsx [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+  // Hadoop-style XML configuration snippet. Selection resolves against the full
+  // row set so it survives search/category filtering.
+  const copyArguments = async () => {
+    const chosen = selectedRowKeys.length
+      ? allRows.filter((r) => selectedRowKeys.includes(r.key))
+      : rows;
+    if (!chosen.length) {
+      return;
+    }
+    await navigator.clipboard.writeText(buildConfigXml(chosen));
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/pages/Overview/sections/JvmSection.tsx]: Should we add error handling around `navigator.clipboard`? It only exists in a secure context, and the OM UI is served over plain HTTP in non-secure clusters.

File: ozone-ui/packages/shared/src/components/ErrorBoundary/QueryErrorBoundary.tsx [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+ * The default fallback: a server-side failure (HTTP 5xx) shows the 500 state;
+ * anything else — a network/timeout failure (native `fetch` rejects with a
+ * `TypeError`), an aborted request, or an unclassified error — shows the network
+ * state. Both wire their action button to `retry`.
+ */
+function defaultFallback({ error, retry }: QueryErrorFallbackProps): React.ReactNode {
+  if (error instanceof HttpError && error.status >= 500) {
+    return <ServerErrorState onAction={retry} />;
+  }
+  return <NetworkErrorState onAction={retry} />;
```
Comment (chihsuan) [inline ozone-ui/packages/shared/src/components/ErrorBoundary/QueryErrorBoundary.tsx]: Should 4xx have its own branch? Right now, everything that is not a 5xx shows **Network Error**, even a render-time TypeError.

File: ozone-ui/packages/shared/src/data/fetchJson.ts [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    headers: {
+      Accept: 'application/json',
+      ...(isJsonBody ? { 'Content-Type': 'application/json' } : {}),
+      ...headers,
+    },
+    body: isJsonBody ? JSON.stringify(body) : (body as BodyInit | null | undefined),
+  });
+
+  if (!response.ok) {
+    throw new HttpError(response.status, url, `${response.status} ${response.statusText}`);
```
Comment (chihsuan) [inline ozone-ui/packages/shared/src/data/fetchJson.ts]: Could we keep the response body on `HttpError`? The status text alone drops whatever the server explained.

File: ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+  const chipLabel = enabled ? 'Live Sync' : 'Manual Sync';
+
+  const handleDbSync = async () => {
+    if (!dbSync || dbSyncing) {
+      return;
+    }
+    setDbSyncing(true);
+    try {
+      await fetchJson(dbSync.url, { method: 'POST' });
+    } finally {
```
Comment (chihsuan) [inline ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx:121]: There is no catch here, so a failed sync closes the dropdown and looks like it worked. Should we surface the error?

File: ozone-ui/packages/om/src/api/useJmx.ts [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+    queryFn: () => queryJmx<T>(qry),
+  };
+}
+
+/**
+ * Fetch a single JMX MBean (the first bean) for a section via TanStack Query.
+ * Requests are de-duplicated by query key, so multiple sections depending on the
+ * same MBean share one network call. Refresh by invalidating the `['jmx']` key.
+ */
+export function useJmxBean<T>(qry: string, options: UseJmxBeanOptions = {}): JmxBeanState<T> {
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/useJmx.ts]: Is this `method` still needed? It looks like nothing calls it, and it does not follow the "Live Sync interval" as the suspense version does.

File: ozone-ui/packages/om/src/api/overview.ts [hunk 16]
```diff
@@ -0,0 +1,327 @@
...
+   * `OMMXBean.getRatisRoles` / `OmUtils.format`). On error the bean returns a
+   * single-element row `[message]`.
+   */
+  RatisRoles: string[][];
+  RatisLogDirectory: string;
+  RocksDbDirectory: string;
+  Version: string;
+  SoftwareVersion: string;
+  StartedTimeInMillis: number;
+  CompileInfo: string;
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/overview.ts]: Just curious, I couldn't find `CompileInfo` on `OMMXBean` or its parent `ServiceRuntimeInfo`. On a real OM the Compiled field renders blank.

https://github.com/apache/ozone/blob/8e872d33a159a65fa06209494fe24c68984c1830/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMMXBean.java#L28-L44

File: ozone-ui/packages/shared/src/components/UtilityBar/UtilityBar.tsx [hunk 17]
```diff
@@ -81,11 +95,16 @@ export const UtilityBar: React.FC<UtilityBarProps> = ({
...
-        color: semanticColors.textDisabled,
+        gap: spacing.sm,
+        color: semanticColors.textSecondary,
       }}
     >
-      {actions}
+      <IconButton
+        icon={<QuestionCircleOutlined style={{ fontSize: 18 }} />}
+        label="Help"
+        onClick={onHelp}
```
Comment (chihsuan) [inline ozone-ui/packages/shared/src/components/UtilityBar/UtilityBar.tsx]: Should the `Help` button render only when `onHelp` is passed? Currently, the button is there, but clicking it does nothing.

https://github.com/user-attachments/assets/7fb9f528-5e8f-43d2-abd2-0ab6fdfdc6ca

**Review by chihsuan — COMMENTED** [review]
Thanks for the updates! @spacemonkd the TanStack Query migration reads really nicely. 👍

I ran this against a real cluster and found a few more issues, mostly around error handling. Please see the inline comments.

Also, one small thing, I noticed a few `gap` and `padding` values written as raw numbers where spacing tokens exist. Worth a quick pass?

File: ozone-ui/packages/om/src/pages/Overview/sections/JvmSection.tsx (hunk 11 above)
Comment (spacemonkd) [reply ozone-ui/packages/om/src/pages/Overview/sections/JvmSection.tsx]: Thanks for pointing this out, I overlooked it.
Got an idea from Claude and switched to using a hidden textarea to support copying over unsecure connection.

But if that is not allowed it fails ultimately and gives proper error. However since Ozone is mostly used in prod I am hoping there aren't a lot of users having unsecured Ozone deployments.

File: ozone-ui/packages/shared/src/components/ErrorBoundary/QueryErrorBoundary.tsx (hunk 12 above)
Comment (spacemonkd) [reply ozone-ui/packages/shared/src/components/ErrorBoundary/QueryErrorBoundary.tsx]: Ideally it should, I didn't have a proper artwork for the 4xx type errors. This is still being worked on so it has been unified under the network error.
Since this is a feature branch we can improve these as I have the updated designs created.

File: ozone-ui/packages/shared/src/data/fetchJson.ts (hunk 13 above)
Comment (spacemonkd) [reply ozone-ui/packages/shared/src/data/fetchJson.ts]: Addressed in the latest change

File: ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx (hunk 14 above)
Comment (spacemonkd) [reply ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx:121]: Addressed in the latest change.

File: ozone-ui/packages/om/src/api/useJmx.ts (hunk 15 above)
Comment (spacemonkd) [reply ozone-ui/packages/om/src/api/useJmx.ts]: Yeah, removed this in the latest commit and also updated the docs to use the new suspense based hook

File: ozone-ui/packages/om/src/api/overview.ts (hunk 16 above)
Comment (spacemonkd) [reply ozone-ui/packages/om/src/api/overview.ts]: Thanks, it was something provided in the mock-ups so I assumed this was present.
I cross checked and it seems it is missing. Probably an idea for a future improvement? Removed it for now

File: ozone-ui/packages/shared/src/components/UtilityBar/UtilityBar.tsx (hunk 17 above)
Comment (spacemonkd) [reply ozone-ui/packages/shared/src/components/UtilityBar/UtilityBar.tsx]: Change the behaviour to only render when onHelp is passed.
Ideally the plan is to redirect this to the Apache Ozone docs or embed them in the site.

**Comment by spacemonkd** [comment]
Thanks for bearing with me @chihsuan. The extensive review is really helpful.
I have addressed the issues, hoping I have not missed anything this time.

Added some more unit tests using Claude and also returning undefined for empty/non-JSON 2xx bodies.
Also `buildJvmHighlights` guards `SystemProperties ?? []` which could have caused an error.

Could you take another look?
Please feel free to let me know if you think we can still improve anything, we can create follow up JIRAs to improve since this is going into the feature branch.

File: ozone-ui/packages/om/src/api/overview.ts [hunk 18]
```diff
@@ -0,0 +1,340 @@
...
+      const [hostName = '', nodeId = '', ratisPort = '', roleRaw = ''] = row;
+      const role = roleRaw.toUpperCase();
+      return {
+        key: nodeId || String(index),
+        hostName,
+        nodeId,
+        ratisPort,
+        role,
+        // The leader has no "readiness"; followers are shown as synced with the leader.
+        readiness: role === 'LEADER' ? null : 'Synced',
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/api/overview.ts:147]: Should this column show the readiness value the bean returns? Right now every follower shows a green **Synced**, so an OM that is not ready still looks healthy.

File: ozone-ui/packages/om/src/pages/Overview/sections/InstanceDetailsSection.tsx [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+
+const kvGridStyle: React.CSSProperties = {
+  display: 'grid',
+  gridTemplateColumns: 'repeat(auto-fill, minmax(200px, 1fr))',
+  gap: `${spacing.lg}px ${spacing.xl}px`,
+};
+
+const InstanceDetailsContent: React.FC = () => {
+  // Fetch all four beans in parallel (avoids an intra-component suspense waterfall).
+  const [omInfoQ, ratisQ, countQ, elapsedQ] = useSuspenseQueries({
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/pages/Overview/sections/InstanceDetailsSection.tsx:43]: Should these two pick up the shared refresh interval? I noticed they only fetch once, so **Election Count** and **Last Election Elapsed Time** stay frozen while the chip still says **Live Sync**.

File: ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx [hunk 20]
```diff
@@ -0,0 +1,253 @@
...
+        </>
+      )}
+    </div>
+  );
+
+  return (
+    <Dropdown
+      open={open}
+      onOpenChange={setOpen}
+      overlay={dropdownContent}
```
Comment (chihsuan) [inline ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx:191]: nit: Worth switching to `popupRender`? `overlay` is on antd's deprecated list. I just see it warning in dev.

```suggestion
      popupRender={() => dropdownContent}
```

https://5x.ant.design/components/dropdown

File: ozone-ui/packages/om/src/navigation.tsx [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+        'Documentation',
+        '/documentation',
+        <BookOutlined style={iconStyle} />
+      ),
+      navItem('log-levels', 'Log levels', '/log-levels', <HistoryOutlined style={iconStyle} />),
+    ],
+  },
+];
+
+/** Product branding shown in the top utility bar. */
```
Comment (chihsuan) [inline ozone-ui/packages/om/src/navigation.tsx:87]: nit: This comment looks like it came from another place. Should it describe the sidebar width, or should it be dropped?

**Review by chihsuan — COMMENTED** [review]
Thanks for including me on this @spacemonkd, and for the patient updates. 🙂 

I left two inline comments and a couple of nits. Other than those it looks good to me.

Three things that might be worth follow-up JIRAs:

1. `Toasts` ignore the theme. Wrapping `ThemeProvider` in antd's `App` would fix all four apps at once.
2. Could we drop `moment`? It is imported once for a single date format, and `SyncChip` already does the same job with native `toLocaleString`.
3. `dev:om` reads `shared/dist`, so shared changes need a rebuild. A dev alias to `shared/src`, or a watch script, would help.

File: ozone-ui/packages/om/src/navigation.tsx (hunk 21 above)
Comment (spacemonkd) [reply ozone-ui/packages/om/src/navigation.tsx]: Dropped the comment. Thanks for pointing it out

File: ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx (hunk 20 above)
Comment (spacemonkd) [reply ozone-ui/packages/shared/src/components/SyncChip/SyncChip.tsx]: Addressed in the latest commit

File: ozone-ui/packages/om/src/pages/Overview/sections/InstanceDetailsSection.tsx (hunk 19 above)
Comment (spacemonkd) [reply ozone-ui/packages/om/src/pages/Overview/sections/InstanceDetailsSection.tsx:46]: Addressed in the latest commit

File: ozone-ui/packages/om/src/api/overview.ts (hunk 18 above)
Comment (spacemonkd) [reply ozone-ui/packages/om/src/api/overview.ts]: Addressed in the latest commit, mapping the READINESS states to more readable format and returning.

**Comment by spacemonkd** [comment]
Thanks @chihsuan for your effort in testing and reviewing this patch.
I have addressed the review comments you gave.
Also I have done the following:
- `ThemeProvider` now wraps children in `<App component={false}>`. `SyncChip` and `JvmContent` use `App.useApp()`'s message instead of the static import so toasts inherit the theme.
- Removed moment from the package.json

Having a watch script / alias to shared components was not handled here. We can file a JIRA and do that separately.
Would you like to do that and pick it up?

**Review by chihsuan — APPROVED** [review]
Thanks for working through all the feedback @spacemonkd +1 LGTM

> Would you like to do that and pick it up?

Sure, happy to pick up the shared-package watch script. 🙂

**Review by ArafatKhan2198 — APPROVED** [review]
Thanks for working on this @spacemonkd 
And thanks for driving the review discussion offline!
LGTM +1
